services/notification/notifiers/comment/__init__.py

```diff
@@ -351,18 +352,36 @@ def is_enabled(self) -> bool:
             self.notifier_yaml_settings, dict
         )
 
+    async def is_first_pull(self, comparison: Comparison) -> bool:
```

Comment:
Possibly a stupid question, but why is this `async`?

Reply:
It does IO to the database and is called in an async context so I figured this should be async as well.  I honestly don't understand how we're doing database queries in other async contexts elsewhere in the code - like why don't we have to `await` the calls into SQLAlchemy?